prereise/gather/hydrodata/eia/tests/test_decompose_profile.py

```diff
@@ -76,7 +77,8 @@ def test_get_profile_by_plant_argument_value():
 
 
 def test_get_profile_by_plant():
-    plant_df = mock_plant_df[mock_plant_df.type == "hydro"]
+    mdf = mock_plant_df.set_index("plant_id")
+    plant_df = mdf[mdf.type == "hydro"]
```

Comment:
Not sure why this test passed for old version of `pandas` maybe it sets the index as the first column by default? Looking through `get_profile_by_plant`, we do need the index of `plant_df` to be `plant_id`, otherwise it will give profile data frame with wrong column names (plant ids). We can still make it one liner though
```python
plant_df = mock_plant_df.set_index("plant_id").query("type == 'hydro'")
```

The reason it fails other tests if we change it at the module level is because `MockGrid` will look for `plant_id` column and set it as index when building up plant data frame of an attribute of the mock grid. Hence it won't work if `plant_id` is already the index.